Library/Homebrew/cmd/info.rb

```diff
@@ -244,7 +244,13 @@ def info_formula(f, args:)
     specs = []
 
     if (stable = f.stable)
-      s = "stable #{stable.version}"
+      latest_version = if ENV["HOMEBREW_JSON_CORE"].present?
+        BottleAPI.latest_pkg_version(f.name).version || stable.version
+      else
+        stable.version
+      end
+
+      s = "stable #{latest_version}"
       s += " (bottled)" if stable.bottled? && f.pour_bottle?
```

Comment:
This may not be the right thing to do here, I'm not sure. Now, `stable.bottled? && f.pour_bottle?` will check using the most recent version of the formula that exists in homebrew/core. This version will become outdated quickly because the user shouldn't need to be updating their homebrew/core tap anymore. That means the `(bottled)` suffice may not necessarily be correct.

Is it better to just scratch the `(bottled)` suffix if we're using `BottleAPI.latest_pkg_version`? Or, maybe instead of `(bottled)` have `(latest)` or something like that? I don't want to give the impression that the formula _isn't bottled_ but also don't want to give the impression the bottle is installable on a particular machine only because an older version was.

Reply:
> Is it better to just scratch the `(bottled)` suffix if we're using `BottleAPI.latest_pkg_version`?

Can this give us the bottled state now/theoretically?

Reply:
What do you mean by "bottled state now/theoretically"? Is "now" the currently installed formula or the latest available formula?

If you mean the latest available formula, I don't think we can check without downloading the bottle

Reply:
But maybe it's okay to download the bottle. Thinking ahead, if we want people to be able to run `brew info` without having homebrew/core tapped, maybe we _should_ make `brew info` download the bottle and use the latest formula for the information

Reply:
Can the bottled state for the latest bottle not be obtained from e.g. https://formulae.brew.sh/api/formula/a2ps.json?

I think `brew info` downloading a bottle feels weird.

Reply:
Kind of. We can definitely check whether a bottle is available for the current OS, but that won't take into account any `pour_bottle?` blocks. That might be fine, though, since those are pretty rare

> I think `brew info` downloading a bottle feels weird.

Note, though, that if homebrew/core is untapped, `brew info` will not work at all

Reply:
> Kind of. We can definitely check whether a bottle is available for the current OS, but that won't take into account any `pour_bottle?` blocks. That might be fine, though, since those are pretty rare

Yeh, I think it's fine to assume that.



> Plus, I forgot that https://formulae.brew.sh/api/formula/a2ps.json exists which we can use in the future to fill out the information instead of needing a bottle

Yeh, that's the output of `brew info --json` anyway so I think we could use this to populate `brew info`. Fine if that's not the case for now 👍🏻